test/TemporaryDirectory.h

```diff
@@ -48,4 +49,19 @@ class TemporaryDirectory
 	boost::filesystem::path m_path;
 };
 
+/**
+ * An object that changes current working directory and restores it upon destruction.
+ */
+class TemporaryWorkingDirectory
```

Comment:
I put the class in `TemporaryDirectory.h` which already contains a similar helper but maybe I should create a separate `TemporaryWorkingDirectory.h` instead?

Reply:
If we have such a granular separation in the test directory, then yes, otherwise I think it is fine.